src/transport.js

```diff
@@ -1,3 +1,32 @@
 import request from 'request-promise';
+import promiseRetry from 'promise-retry';
 
-export default request;
+// When the API sends a 429 back, retry it for 10 times
+// This code is mainly copied from
+// https://github.com/CanTireInnovations/request-promise-retry/blob/master/index.js
+
+const retryRequest = (promiseRequest) => {
+  return req => {
```

Comment:
`(req, retryOptions) => promiseRetry(...`

And later on:
`Object.assign(defaults, retryOptions);`